### ISSUE TITLE
New feature son-sec-gw

### ISSUE THREAD

**Issue by felipevicens** [sonata-nfv/son-gkeeper#434]
@jbonnet Do you think this is the repo that should contains the component son-sec-gw?

The proposal is the following:

```

              +----------------+                +---------------+
     https    |                |     http       |               |
   +---------^+  son-sec-gw    +---------------->  son-gtk-api  |
              |                |        |       |               |
              +----------------+        |       +---------------+
                                        |
                                        |       +---------------+
                                        |       |               |
                                        +------->    son-gui    |
                                                |               |
                                                +---------------+
```

**son-sec-gw** is a nginx proxy, which is listen in port 32001 forwarding traffic to son-gtkapi and listen in port 80 forwarding traffic to son-gui component.

**son-sec-gw** will have the https certificate mounted as docker volume.

**Comment by jbonnet** [comment]
Perfect!
Go ahead!

**Comment by felipevicens** [comment]
@jbonnet currently we have:

http:
son-gtkapi -> 32001
son-gui -> 80

We will have:
https:
son-gui -> 443

What about son-gtkapi? Do we change 32001 to only https or should we create a new port for https gtkapi?

**Comment by jbonnet** [comment]
But isn’t the idea to keep everything behing the SEC-GTW the same?
By having chosen the port to distinguish between different apps (GUI vs. API), we’re now choke…
For example, GitHub’s API can be accessed by

  `$ curl https://api.github.com/users/jbonnet` <https://api.github.com/users/jbonnet%60>

Could we have the same, or is it too late to change (in terms of effort, I mean)?
Then the URL could be simplified to `…/v2/…`(droping the current `…/api` prefix. 

With less impact would be something like

  `$ curl https://sp.sonata-nfv.eu/api` <https://sp.sonata-nfv.eu/api%60>

for the GK’s API (with HTTP access simply being redirected to HTTPS).

The GUI could be accessed through HTTP (port `80`), being redirected to HTTPS (port 443) automatically by NGINx.

What do you think?

**Comment by felipevicens** [comment]
Sounds great! I'll try

```
son-gui --> /
son-gtkapi --> /api

```

**Comment by jbonnet** [comment]
Perfect!
;-)

**Comment by felipevicens** [comment]
@jbonnet If I point /api from nginx to -> /api in gtkapi we will loose / that have the API swagger definition. Could you move / to /api page? Also I think this need to be updated adding /api/v2 in the paths

This is:

```
---
  swagger: "2.0"
  info: 
    title: "SONATA's SP API"
    description: "This is a proposal for the SONATA's SP API"
    version: "1.0.0"
  host: "sp.sonata-nfv.eu"
...
...
```

**Comment by jbonnet** [comment]
Sure, no problem.

**Comment by felipevicens** [comment]
@jbonnet I've configured the following workflow for https support:

Case 1: HTTPS Certificate is present
listen port 80 --> Redirect 301 to https
listen port 443 --> / --> gui and /api -> gtkapi

Case 2: HTTPS Certificate is NOT present
listen port 80 --> / --> gui and /api -> gtkapi
no 443

**Comment by jbonnet** [comment]
Sounds good!

**Comment by felipevicens** [comment]
It is working and available in version 2.1
